src/datasource.test.ts

```diff
@@ -1232,6 +1234,176 @@ describe('OpenSearchDatasource', function(this: any) {
       );
     });
   });
+  describe('#executeLuceneQueries', () => {
```

Comment:
Tried to test this with [marble diagrams and rxjs TestRunner](https://rxjs-dev.firebaseapp.com/guide/testing/marble-testing), but couldn't get the runtime observables to match the diagrams :( I also had to turn the observables to promises since as far as I could find, that's the easiest way to test the results of the calls in jest. This is why I only ever put two query types (that result in only two observables that I could turn into promise with firstValueFrom and lastValueFrom) in the tests.